src/main/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesService.java

```diff
@@ -1,64 +1,40 @@
 package uk.gov.digital.ho.pttg.application.namematching;
 
-import com.google.common.collect.Lists;
-import org.apache.commons.lang3.StringUtils;
 import org.springframework.stereotype.Service;
 import uk.gov.digital.ho.pttg.application.namematching.candidates.MultipleLastNames;
 import uk.gov.digital.ho.pttg.application.namematching.candidates.NameCombinations;
+import uk.gov.digital.ho.pttg.application.namematching.candidates.SpecialCharacters;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 import static uk.gov.digital.ho.pttg.application.namematching.NameMatchingFunctions.deduplicate;
 
 @Service
 public class NameMatchingCandidatesService {
-    private static final String NAME_SPLITTERS = "-'";
-    private static final String NAME_SPLITTER_REGEX = "[" + NAME_SPLITTERS + "]";
 
     private NameCombinations nameCombinations;
     private MultipleLastNames multipleLastNames;
+    private SpecialCharacters specialCharacters;
```

Comment:
Should the type of each of these be NameMatchingGenerator since that's the interface?